rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -65,6 +67,15 @@ foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
     list(APPEND _generated_srv_py_files
       "${_output_path}/${_parent_folder}/_${_module_name}.py"
     )
+  elseif(_parent_folder STREQUAL "action")
+    if("_${_module_name}_s.c" MATCHES "(.*)__response(.*)" OR "_${_module_name}_s.c" MATCHES "(.*)__request(.*)")
```

Comment:
This works for services, but not messages. After adding `action/Something.msg` in test_msgs I see this traceback

`undefined symbol: test_msgs__action__something__convert_from_py`

```
342: (test_publisher) pid 7968: ['/usr/bin/python3', '/home/developer/workspaces/ros2-dev0/src/ros2/system_tests/test_communication/test/publisher_py.py', 'BoundedArrayPrimitives', '/test_time_23_08_40'] (stderr > stdout, all > console)
342: (test_subscriber) pid 7969: ['/home/developer/workspaces/ros2-dev0/build/test_communication/test_subscriber_cpp', 'BoundedArrayPrimitives', '/test_time_23_08_40'] (stderr > stdout, all > console)
342: [test_publisher] exception in talker:
342: [test_publisher] Traceback (most recent call last):
342: [test_publisher]   File "/home/developer/workspaces/ros2-dev0/install/rosidl_generator_py/lib/python3.6/site-packages/rosidl_generator_py/import_type_support_impl.py", line 42, in import_type_support
342: [test_publisher]     return importlib.import_module(module_name, package=pkg_name)
342: [test_publisher]   File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
342: [test_publisher]     return _bootstrap._gcd_import(name[level:], package, level)
342: [test_publisher]   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
342: [test_publisher]   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
342: [test_publisher]   File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
342: [test_publisher]   File "<frozen importlib._bootstrap>", line 658, in _load_unlocked
342: [test_publisher]   File "<frozen importlib._bootstrap>", line 571, in module_from_spec
342: [test_publisher]   File "<frozen importlib._bootstrap_external>", line 922, in create_module
342: [test_publisher]   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
342: [test_publisher] ImportError: /home/developer/workspaces/ros2-dev0/install/test_msgs/lib/python3.6/site-packages/test_msgs/test_msgs_s__rosidl_typesupport_c.cpython-36m-x86_64-linux-gnu.so: undefined symbol: test_msgs__action__something__convert_from_py
```

Reply:
Changing this condition gets a little further,

```cmake
    get_filename_component(_ext "${_idl_file}" EXT)
    # C files generated for <msg>.msg, <service>_Request.msg and <service>_Response.msg but not <service>.srv
    if(_ext STREQUAL ".msg")
      list(APPEND _generated_action_c_files
        "${_output_path}/${_parent_folder}/_${_module_name}_s.c"
      )
    endif()
    list(APPEND _generated_action_py_files
      "${_output_path}/${_parent_folder}/_${_module_name}.py"
    )
```


Though `test_communication` still fails with 
```
18: [test_publisher] exception in talker:
18: [test_publisher] Traceback (most recent call last):
18: [test_publisher]   File "/home/developer/workspaces/ros2-dev0/src/ros2/system_tests/test_communication/test/publisher_py.py", line 63, in <module>
18: [test_publisher]     number_of_cycles=args.number_of_cycles)
18: [test_publisher]   File "/home/developer/workspaces/ros2-dev0/src/ros2/system_tests/test_communication/test/publisher_py.py", line 27, in talker
18: [test_publisher]     msg_mod = getattr(module, message_name)
18: [test_publisher] AttributeError: module 'test_msgs.msg' has no attribute 'Something'
18: [test_subscriber] exception in subscriber:
18: [test_subscriber] Traceback (most recent call last):
18: [test_subscriber]   File "/home/developer/workspaces/ros2-dev0/src/ros2/system_tests/test_communication/test/subscriber_py.py", line 81, in <module>
18: [test_subscriber]     listener(message_name=args.message_name, namespace=args.namespace)
18: [test_subscriber]   File "/home/developer/workspaces/ros2-dev0/src/ros2/system_tests/test_communication/test/subscriber_py.py", line 47, in listener
18: [test_subscriber]     msg_mod = getattr(module, message_name)
18: [test_subscriber] AttributeError: module 'test_msgs.msg' has no attribute 'Something'
```
but I'm not yet sure if the problem is here or there

Reply:
Fixed in 98d630c5d43c1395525e5a2b23596b3ff3b4b2c9